heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/WebSink.java

```diff
@@ -0,0 +1,162 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricsmgr.sink;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import com.twitter.heron.common.basics.TypeUtils;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsInfo;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsRecord;
+import com.twitter.heron.spi.metricsmgr.sink.IMetricsSink;
+import com.twitter.heron.spi.metricsmgr.sink.SinkContext;
+
+
+/**
+ * A metrics sink that publishes metrics on a http endpoint
+ */
+public class WebSink implements IMetricsSink {
+  private static final Logger LOG = Logger.getLogger(WebSink.class.getName());
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private static final int HTTP_STATUS_OK = 200;
+
+  private static final String KEY_PORT = "port";
+  private static final String KEY_PORT_FILE = "port-file";
+  private static final String KEY_PATH = "path";
+  private static final String KEY_FLAT_METRICS = "flat-metrics";
+  private static final String KEY_INCLUDE_TOPOLOGY_NAME = "include-topology-name";
+
+  protected ConcurrentHashMap<String, Object> metrics = new ConcurrentHashMap<>();
+  private HttpServer httpServer;
+  private boolean isFlatMetrics = true;
+  private boolean includeServiceNamespace = true;
+  private String topologyName;
+
+  @Override
+  public void init(Map<String, Object> conf, SinkContext context) {
+    String path = (String) conf.get(KEY_PATH);
+    String portFile = (String) conf.get(KEY_PORT_FILE);
+    isFlatMetrics = TypeUtils.getBoolean(conf.getOrDefault(KEY_FLAT_METRICS, true));
+    includeServiceNamespace = TypeUtils.getBoolean(
+        conf.getOrDefault(KEY_INCLUDE_TOPOLOGY_NAME, false));
+    topologyName = context.getTopologyName();
+
+    int port = TypeUtils.getInteger(conf.getOrDefault(KEY_PORT, 0));
+    if (port == 0) {
+      if (portFile != null) {
+        try {
+          port = TypeUtils.getInteger(Files.lines(Paths.get(portFile)).findFirst().get().trim());
+        } catch (IOException e) {
+          throw new IllegalArgumentException("Could not parse " + KEY_PORT_FILE + " " + portFile
```

Comment:
TypeUtils.getInteger already does that. However I should probably catch and re-throw that to add context.